controllers/workspace/devworkspace_controller_test.go

```diff
@@ -602,6 +602,85 @@ var _ = Describe("DevWorkspace Controller", func() {
 		})
 	})
 
+	Context("DevWorkspace deployment", func() {
+		const testURL = "test-url"
+
+		BeforeEach(func() {
+			workspacecontroller.SetupHttpClientsForTesting(&http.Client{
+				Transport: &testutil.TestRoundTripper{
+					Data: map[string]testutil.TestResponse{
+						fmt.Sprintf("%s/healthz", testURL): {
+							StatusCode: http.StatusOK,
+						},
+					},
+				},
+			})
+		})
+
+		AfterEach(func() {
+			deleteDevWorkspace(devWorkspaceName)
+			workspacecontroller.SetupHttpClientsForTesting(getBasicTestHttpClient())
+		})
+
+		It("Sets the runtimeClassName from the DWOC", func() {
+			By("Creating DWOC with a runtimeClassName")
+			runtimeClassName := "test-runtime-class"
+			dwoc := generateDwoc("devworkspace-operator-config")
+			dwoc.Config = &controllerv1alpha1.OperatorConfiguration{
+				Workspace: &controllerv1alpha1.WorkspaceConfig{
+					RuntimeClassName: &runtimeClassName,
+				},
+			}
+
+			createObject(dwoc)
+			defer deleteObject(dwoc)
+
+			createDevWorkspace(devWorkspaceName, "test-devworkspace-config.yaml")
+			devworkspace := getExistingDevWorkspace(devWorkspaceName)
+			workspaceID := devworkspace.Status.DevWorkspaceId
+
+			By("Manually making Routing ready to continue")
+			markRoutingReady(testURL, common.DevWorkspaceRoutingName(workspaceID))
+
+			deploy := &appsv1.Deployment{}
+			deployNN := namespacedName(common.DeploymentName(workspaceID), testNamespace)
+			Eventually(func() error {
+				return k8sClient.Get(ctx, deployNN, deploy)
+			}, timeout, interval).Should(Succeed(), "Getting workspace deployment from cluster")
+			Expect(*deploy.Spec.Template.Spec.RuntimeClassName).Should(Equal("test-runtime-class"))
+		})
+
+		It("Sets the runtimeClassName from the attribute instead of the DWOC", func() {
+			By("Creating DWOC with a runtimeClassName")
+			runtimeClassName := "test-runtime-class"
+			dwoc := generateDwoc("devworkspace-operator-config")
+			dwoc.Config = &controllerv1alpha1.OperatorConfiguration{
+				Workspace: &controllerv1alpha1.WorkspaceConfig{
+					RuntimeClassName: &runtimeClassName,
+				},
+			}
+
+			createObject(dwoc)
+			defer deleteObject(dwoc)
+
+			createDevWorkspace(devWorkspaceName, "test-devworkspace-config.yaml")
+			devworkspace := getExistingDevWorkspace(devWorkspaceName)
+			devworkspace.Spec.Template.Attributes.PutString(constants.RuntimeClassNameAttribute, "test-runtime-class-attribute")
+			updateDevWorkspace(devworkspace)
```

Comment:
I suggest using an `Eventually()` here so that we can set a timeout for the updateDevWorkspace function, similar to what we are [doing](https://github.com/devfile/devworkspace-operator/blob/8fc183f55f20e6854f220a2c4bb844190366f67a/controllers/workspace/devworkspace_controller_test.go#L634) elsewhere.

Additionally, since `updateDevWorkspace()` is not called anywhere else, you could just write it out inline rather than have a function for it. Though, having a function that you call after modifying the devworkspace seems to make sense :)

Reply:
Updated, and removed `updateDevWorkspace()`